resources/js/requests/components/FilterMobile.vue

```diff
@@ -0,0 +1,217 @@
+              <i class="fas fa-caret-down"></i>
+            </button>
+            <div
+              class="dropdown-menu"
+              aria-labelledby="dropdownMenuButton"
+            >
+              <a
+                class="dropdown-item"
+                href="#"
+                @click="selectOption('By Due Date', 'orderBy')"
+              >
+                By Due Date</a
+              >
+              <a
+                class="dropdown-item"
+                href="#"
+                @click="selectOption('By Creation Date', 'orderBy')"
+              >
+                By Creation Date</a
+              >
+            </div>
+          </div>
+
+          <div>
+            <div class="input-group">
+              <input
+                type="text"
+                class="form-control narrow-input"
+                v-model="searchCriteria"
+                placeholder="(fulltext LIKE '%someText%')"
+              />
+              <div class="input-group-append">
+                <button
+                  class="btn btn-primary"
+                  @click="performSearch"
+                >
+                  <i class="fas fa-search"></i>
+                </button>
+              </div>
+            </div>
+          </div>
+        </b-nav>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+export default {
+  props: {
+    type: String,
+  },
+  data() {
+    return {
+      searchCriteria: "",
+      selectedOption: "In Progress",
+      apiData: [],
+      items: [],
+      orderBy: "id",
+      orderDirection: "DESC",
+      additionalParams: "",
+      sortOrder: [
+        {
+          field: "id",
+          sortField: "id",
+          direction: "desc",
+        },
+      ],
+      fields: [],
+      previousFilter: "",
+      previousPmql: "",
+    };
+  },
+  methods: {
+    selectOption(option, controlName) {
+      let apiPath = buildApiPath(option, controlName);
+      callApiFilter(apiPath);
+    },
+    buildapiPath(option, controlName) {
+      let basePath =
+        this.type + "?page=1&per_page=10&include=process,participants,data&";
+      if (controlName === "status") {
+        return basePath + 'pmql=(status = "' + option + '")';
+      } else if (controlName === "filter") {
+        return basePath + 'filter = "' + option + '"';
+      } else if (controlName === "orderBy") {
+        return basePath + 'order_by = "' + option + '"';
+      } else if (controlName === "search") {
+        return basePath + 'pmql=(fulltext LIKE "%' + option + '%")';
+      }
+    },
+    callApiFilter(apiPath) {
+      ProcessMaker.apiClient
+        .get(apiPath, { baseURL: "" })
+        .then((response) => {
+          this.apiData = response.data;
+        })
+        .catch((error) => {
+          console.error("Error calling API:", error);
+        });
+    },
+    performSearch() {
+      let apiPath = buildApiPath(this.searchCriteria, "search");
+      callApiFilter(apiPath);
```

Comment:
to call a method use this.callApiFilter
instead of create apiPath set it  this.callApiFilter(this.buildApiPath(this.searchCriteria, "search"))

Reply:
solved